frontend/src/components/AddMember/Health.js

```diff
@@ -1,45 +1,55 @@
 import { ListHeader } from "@code4ro/taskforce-fe-components";
 import SelectList from "../SelectList";
+import { mapPreExistMedCondValues } from "../../utils";
 import { options } from "./options";
-import React from "react";
+import React, { useEffect } from "react";
 import PropTypes from "prop-types";
 
-const Health = ({ userData, setUserDataField, titles }) => (
-  <>
-    <ListHeader title={titles.healthInfo} />
-    <ListHeader title={titles.smoker} />
-    <SelectList
-      options={options.yesNo}
-      name="smoker"
-      required={true}
-      value={[userData.smoker]}
-      onChange={([value]) => setUserDataField("smoker", value)}
-    />
-    <ListHeader title={titles.preexistingConditions} />
-    <SelectList
-      options={options.preexistingMedicalCondition}
-      name="preexistingMedicalCondition"
-      value={
+const Health = ({ userData, setUserDataField, titles }) => {
+  // TODO: We need to rethink a bit on how to
+  // properly do the data mapping for this component
+  useEffect(() => {
+    if (userData.preexistingMedicalCondition) {
+      const data = mapPreExistMedCondValues(
         userData.preexistingMedicalCondition
-          ? userData.preexistingMedicalCondition
-          : []
-      }
-      multiple={true}
-      onChange={value => {
-        setUserDataField("preexistingMedicalCondition", value);
-      }}
-    />
-  </>
-);
+      );
+      setUserDataField("preexistingMedicalCondition", data);
+    }
+  }, []); // eslint-disable-line
```

Comment:
Why the `// eslint-disable-line` ?

Reply:
It's because we want to trigger the effect only once, and having the below version, it will trigger multiple times. Therefore the `eslint-disable` rule is disabled.
```suggestion
  }, [userData.preexistingMedicalCondition]);
```